pkg/compact/compact.go

```diff
@@ -393,6 +412,8 @@ func NewGroup(
 		blocksMarkedForDeletion:     blocksMarkedForDeletion,
 		blocksMarkedForNoCompact:    blocksMarkedForNoCompact,
 		hashFunc:                    hashFunc,
+		numberOfIterations:          numberOfIterations,
+		numberOfBlocksToMerge:       numberOfBlocksToMerge,
```

Comment:
To me, I prefer adding these progress metrics to the `DefaultPlanSim` struct with label `group`. The reason is:
1. The new compactor progress feature will be possible under a feature flag and won't be enabled by default. Then in this case, we don't want to initialize them in `Group` and need additonal flags to check. Adding it as part of the `DefaultPlanSim` struct makes this control easier.

2. Let's also wrap these progress related metric to a struct called `ProgressMetrics`? Similar to `CompactMetrics`. I can see we might need to add more progress metrics in the future like `numOfSeriesToMerge` or `numOfSamplesToMerge`, etc

Reply:
Yes, since it will eventually be a flag, this is better. 
I've added the `ProgressMetrics` struct in my latest commit. Will update the metrics next. 
Thanks!